internal/hcp/api/service_version.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"context"
+	"fmt"
+
+	hcpPackerAPI "github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/stable/2023-01-01/client/packer_service"
+	hcpPackerModels "github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/stable/2023-01-01/models"
+)
+
+const incompleteVersionName = "v0"
+
+// IsVersionComplete returns if the given version is completed or not.
+//
+// The best way to know if the version is completed or not is from the name of the version. All version that are
+// incomplete are named "v0".
+func (c *Client) IsVersionComplete(version *hcpPackerModels.HashicorpCloudPacker20230101Version) bool {
+	return version.Name != incompleteVersionName
+}
+
+func (c *Client) CreateVersion(
+	ctx context.Context,
+	bucketName,
+	fingerprint string,
+	templateType hcpPackerModels.HashicorpCloudPacker20230101TemplateType,
+) (*hcpPackerAPI.PackerServiceCreateVersionOK, error) {
+
+	params := hcpPackerAPI.NewPackerServiceCreateVersionParamsWithContext(ctx)
+	params.LocationOrganizationID = c.OrganizationID
+	params.LocationProjectID = c.ProjectID
+	params.BucketName = bucketName
+	params.Body = &hcpPackerModels.HashicorpCloudPacker20230101CreateVersionBody{
+		Fingerprint:  fingerprint,
+		TemplateType: templateType.Pointer(),
```

Comment:
This should be the string value not the address
```suggestion
		TemplateType: templateType,
```